webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/GoogleCloudDataVault.scala

```diff
@@ -72,6 +73,15 @@ class GoogleCloudDataVault(uri: URI, credential: Option[GoogleServiceAccountCred
     } yield (bytes, encoding)
   }
 
+  override def listDirectory(path: VaultPath, maxItems: Int)(implicit ec: ExecutionContext): Fox[List[VaultPath]] = {
+    val objName = path.toUri.getPath.tail
+    val blobs = storage.list(bucket, Storage.BlobListOption.prefix(objName), Storage.BlobListOption.currentDirectory())
+    val subDirectories = blobs.getValues.asScala.toList.filter(_.isDirectory).take(maxItems)
+    val paths = subDirectories.map(dirBlob =>
+      new VaultPath(new URI(s"${uri.getScheme}://$bucket/${dirBlob.getBlobId.getName}"), this))
```

Comment:
can any of these synchronous functions throw exceptions? If so, please wrap them (or the whole block) in a tryo. Then, the Fox.successful can be removed (as tryo will create a Box, which is implicitly converted to Fox)